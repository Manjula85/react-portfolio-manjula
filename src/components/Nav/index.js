import React from "react";

function Nav() {
  const categories = [
    {
      name: "About",
      description: "A breif intro about me",
    },
    {
      name: "Portfolio",
      description: "All the projects I have done so far",
    },
    {
      name: "Contact",
      description: "A form to contact me",
    },
    {
      name: "Resume",
      description: "Downloadable qualifications",
    },
  ];

  function categorySelected(name) {
    console.log(`${name} clicked`);
  }

  return (
    <header>
      <h1>Manjula</h1>
      <nav>
        <ul className="flex-row">
          <li>
            <h1><a data-testid="link" href="/">Oh Snap!</a></h1>
          </li>
          <li>
            <h2><a data-testid="about" href="#about">About me</a></h2>
          </li>
          {categories.map((category) => (
            <li className="mx-1" key={category.name}>
              <span onClick={categorySelected(category.name)}>
                {category.name}
              </span>
            </li>
          ))}
        </ul>
      </nav>
    </header>
  );
}

export default Nav;
