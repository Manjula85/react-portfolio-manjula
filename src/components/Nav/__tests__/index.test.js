import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

afterEach(cleanup);

const categories = [
  { name: "portraits", description: "Portraits of people in my life" },
];
const mockCurrentCategory = jest.fn();
const mockSetCurrentCategory = jest.fn();

describe("Nav component", () => {
  //baseline test
  it("renders", () => {
    render(
      <Nav
        categories={categories}
        setCurrentCategory={mockCurrentCategory}
        currentCategory={mockSetCurrentCategory}
      />
    );
  });

  //snapshot test
  it("matches snapshot", () => {
    const { asFragment } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockCurrentCategory}
        currentCategory={mockSetCurrentCategory}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("links are visible", () => {
  it("insert text into the links", () => {
    const { getByTestId } = render(
      <Nav
        categories={categories}
        setCurrentCategory={mockCurrentCategory}
        currentCategory={mockSetCurrentCategory}
      />
    );
    expect(getByTestId("link")).toHaveTextContent("Oh Snap!");
    expect(getByTestId("about")).toHaveTextContent("About me");
  });
});
