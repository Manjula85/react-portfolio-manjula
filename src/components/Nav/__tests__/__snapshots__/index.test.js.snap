// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Manjula
    </h1>
    <nav>
      <ul
        class="flex-row"
      >
        <li>
          <h1>
            <a
              data-testid="link"
              href="/"
            >
              Oh Snap!
            </a>
          </h1>
        </li>
        <li>
          <h2>
            <a
              data-testid="about"
              href="#about"
            >
              About me
            </a>
          </h2>
        </li>
        <li
          class="mx-1"
        >
          <span>
            About
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Portfolio
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            Resume
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
